src/core/exec-invoke.c

```diff
@@ -564,11 +564,8 @@ static int setup_output(
         case EXEC_OUTPUT_JOURNAL_AND_CONSOLE:
                 r = connect_logger_as(context, params, o, ident, fileno, uid, gid);
                 if (r < 0) {
-                        log_exec_warning_errno(context,
```

Comment:
The commit message doesn't paint the full picture. The change to `LOG_CONTEXT_PUSH_EXEC` applies to *all* log lines, incl. those made recursively. The existing calls to `log_exec_*` will be fine, but what about all the other calls?

Reply:
I think applying this to all the recursive log lines is a **good** thing, that's why it's a context, you can apply it to library log messages without those log messages having to know anything about the context.

I actually **want** to know that deep down in some library a log message was generated while spawning unit xxx instead of having that log message without having any information attached on the context it was generated in.

Reply:
I agree with @DaanDeMeyer. Any call into basic libraries is still inside the generic execution _context_, hence they should be marked as so and the same log level applies.

Reply:
Fine. But then this needs to be described in the commit message. Right now it looks like an unintended side effect.

Reply:
OK, this makes sense. But please update the commit message to describe this.

Reply:
Extended the log message